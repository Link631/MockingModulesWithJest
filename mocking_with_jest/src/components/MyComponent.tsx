import React from "react";
import {Modal} from "@fluentui/react";
export interface IMyComponentProps{
    text: string;
}

export class MyComponent extends React.Component<IMyComponentProps, any> {
    render(): JSX.Element {
        return <Modal isOpen={true}  >
        text from props:
            {this.props.text}
        </Modal>
   }
}