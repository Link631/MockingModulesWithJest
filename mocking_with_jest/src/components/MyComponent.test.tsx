import React from 'react';
import { render } from '@testing-library/react';
import {MyComponent} from './MyComponent';
import { mount, configure } from 'enzyme';
import ReactSixteenAdapter from "enzyme-adapter-react-16";

configure({adapter: new ReactSixteenAdapter()})
it('renders the modal', () => {

    const acutalModule = require.requireActual("@fluentui/react");
    const dialogMock = require("./mocks/DialogMock");
    acutalModule.Modal = dialogMock.Modal;

    jest.mock("@fluentUi/react", () => {
        return acutalModule
    }) 
   // test
   const myComponent = mount(<MyComponent text={"some text"}/>);

   // compare
   expect(myComponent.html()).toMatchSnapshot();
});
