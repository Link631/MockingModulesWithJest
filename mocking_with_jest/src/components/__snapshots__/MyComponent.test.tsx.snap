// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the modal 1`] = `"<div>text from props:some text</div>"`;
